src/ucp/core/ucp_listener.c

```diff
@@ -245,35 +316,58 @@ ucs_status_t ucp_listener_create(ucp_worker_h worker,
         iface_params.mode.sockaddr.listen_sockaddr  = params->sockaddr;
         iface_params.mode.sockaddr.cb_flags         = UCT_CB_FLAG_ASYNC;
 
+        if (port) {
+            /* Set the port for the next sockaddr iface. This port was either
+             * obtained from the user or generated by the first created sockaddr
+             * iface if the port from the user was equal to zero */
+            status = ucs_sockaddr_set_port((struct sockaddr *)
+                                           iface_params.mode.sockaddr.listen_sockaddr.addr,
+                                           port);
+            if (status != UCS_OK) {
+                ucs_error("failed to set port %d on iface %s",
```

Comment:
this failure is not really related to the iface
"failed to set sockaddr port to %d"

Reply:
if there are several listening ifaces, wouldn't it help to know which one had the error with the port setting?

Reply:
how is ucs_sockaddr_set_port related to the iface?

Reply:
it sets the port that the iface would listen on. so the message would indicate which iface we are on in the current iteration of the loop.

Reply:
"failed to set port parameter for creating %s iface"